Elements/src/Geometry/CsgExtensions.cs

```diff
@@ -88,23 +89,37 @@ internal static void Tessellate(this Csg.Solid csg, ref Mesh mesh, Transform tra
                 var a = tess.Vertices[tess.Elements[0]].Position.ToVector3();
                 var b = tess.Vertices[tess.Elements[1]].Position.ToVector3();
                 var c = tess.Vertices[tess.Elements[2]].Position.ToVector3();
-                var n = (b - a).Cross(c - a).Unitized();
+                var tmp = (b - a).Unitized();
+                var n = tmp.Cross(c - a).Unitized();
+
+                // Calculate the texture space basis vectors
+                // from the first triangle. This is acceptable
+                // for planar faces.
+                // TODO: Update this when we support non-planar faces.
+                // https://gamedev.stackexchange.com/questions/172352/finding-texture-coordinates-for-plane
+                e1 = n.Cross(n.IsParallelTo(Vector3.XAxis) ? Vector3.YAxis : Vector3.XAxis).Unitized();
```

Comment:
I see this usage pattern repeats, and it comes with a special note. Would it be worth extracting it to a `Vector3` method with a disclaimer in the docs?

```
var ( e1, e2 ) = Vector3.ComputeBasis(n)
```

We could test just the planar cases, for now, then add non-planar when necessary.

Reply:
That is a nice suggestion. I'm going to do it.